tasks/extract-binaries-from-image/extract-binaries-from-image.yaml

```diff
@@ -0,0 +1,71 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: extract-binaries-from-image
+  labels:
+    app.kubernetes.io/version: "1.0.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that extracts binaries to be released on github.com from an image
+  params:
+    - name: image_binaries_path
+      type: string
+      description: Path inside the image where the binaries to extract are stored
+      default: "releases"
+    - name: snapshotPath
+      type: string
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      default: snapshot.json
+    - name: subdirectory
+      description: Subdirectory inside the workspace to be used for storing the binaries
+      type: string
+      default: ""
+  results:
+    - name: binaries_path
+      type: string
+      description: The directory inside the workspace where the binaries are stored
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot is stored. The extracted binaries will be stored here as well.
+  steps:
+    - name: extract-binaries-from-image
+      image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+      script: |
+        #!/bin/sh -ex
+
+        SNAPSHOT_SPEC_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_SPEC_FILE}" ] ; then
+            echo "Error: No valid snapshot file was provided."
+            exit 1
+        fi
+        IMAGE_URL=$(jq -r '.components[0].containerImage // ""' "${SNAPSHOT_SPEC_FILE}")
```

Comment:
I mean this is fine, but I am not sure why a default is added here instead of just 
```
        IMAGE_URL=$(jq -r '.components[0].containerImage' "${SNAPSHOT_SPEC_FILE}")
        if [ "${IMAGE_URL}" == "null" ] ; then

Reply:
I guess if we're confident that it will really be that, then sure. What I still haven't figured out is why `jq` sometimes returns `""` for missing values and sometimes `"null"`. So that's why personally I feel more confident using `// ""`.

Even in their docs they have this example:
```
Command	jq '.foo'
Input	{"notfoo": true, "alsonotfoo": false}
Output	null
```
https://jqplay.org/jq?q=.foo&j=%7B%22notfoo%22%3A%20true%2C%20%22alsonotfoo%22%3A%20false%7D
On the tool, it shows that the output is `null`. Then why does this print an empty string instead?
```sh
jq -r .foo <<< '{"notfoo": true, "alsonotfoo": false}'
```
But this does print null:
```sh
echo $(jq -r .foo <<< '{"notfoo": true, "alsonotfoo": false}')
null
```
So again, it's probably ok to do as you say, but I still prefer the explicit default value.

I'll leave it up to Ernesto :)

Reply:
I'll go with @mmalina's

Reply:
thanks for the explanation Martin